guarddog/analyzer/sourcecode/__init__.py

```diff
@@ -7,27 +7,43 @@
 from guarddog.ecosystems import ECOSYSTEM
 
 current_dir = pathlib.Path(__file__).parent.resolve()
-rule_file_names = list(
+semgrep_rule_file_names = list(
     filter(
         lambda x: x.endswith('yml'),
         os.listdir(current_dir)
     )
 )
 
-SOURCECODE_RULES = {
+SEMGREP_SOURCECODE_RULES = {
     ECOSYSTEM.PYPI: list(),
     ECOSYSTEM.NPM: list(),
 }  # type: dict[ECOSYSTEM, list[dict]]
 
-for file_name in rule_file_names:
+for file_name in semgrep_rule_file_names:
     with open(os.path.join(current_dir, file_name), "r") as fd:
         data = yaml.load(fd, Loader=SafeLoader)
         for rule in data["rules"]:
             for lang in rule["languages"]:
                 match lang:
                     case "python":
-                        if rule not in SOURCECODE_RULES[ECOSYSTEM.PYPI]:
-                            SOURCECODE_RULES[ECOSYSTEM.PYPI].append(rule)
+                        if rule not in SEMGREP_SOURCECODE_RULES[ECOSYSTEM.PYPI]:
+                            SEMGREP_SOURCECODE_RULES[ECOSYSTEM.PYPI].append(rule)
                     case "javascript" | "typescript" | "json":
-                        if rule not in SOURCECODE_RULES[ECOSYSTEM.NPM]:
-                            SOURCECODE_RULES[ECOSYSTEM.NPM].append(rule)
+                        if rule not in SEMGREP_SOURCECODE_RULES[ECOSYSTEM.NPM]:
+                            SEMGREP_SOURCECODE_RULES[ECOSYSTEM.NPM].append(rule)
+
+yara_rule_file_names = list(
```

Comment:
can we document the mechanism to load Yara rules and an example? (in the README and with a comment here)

Reply:
done in [d10b480](https://github.com/DataDog/guarddog/pull/401/commits/d10b4805491a5e569195f256bd2dc9203483799c) [fbb0648](https://github.com/DataDog/guarddog/pull/401/commits/fbb0648ba38319b11d7cbaf8060ed4689d92a230)